QueryEngine/RelAlgAbstractInterpreter.h

```diff
@@ -480,7 +480,6 @@ class RelAlgNode {
   virtual ~RelAlgNode() {}
 
   void setContextData(const void* context_data) const {
-    CHECK(!context_data_);
```

Comment:
I was able to remove this in my testing and not see any asserts. Is this removal required?

Reply:
Yeah I think so, this is where it failed first but the fix in RelAlgExecutor.cpp prevents this from happening.

Reply:
Do you mind putting the check back in place? I am a bit worried about leaking `context_data_` if it wasn't properly cleaned up (though I think right now it only comes from the `row_set_memory_owner`, which should be cleaned up, but I'm not 100% sure on that one).